module/renderer/src/commonMain/kotlin/org/cru/godtools/shared/renderer/content/RenderTabs.kt

```diff
@@ -0,0 +1,111 @@
+@file:OptIn(ExperimentalMaterial3Api::class)
+
+package org.cru.godtools.shared.renderer.content
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.SecondaryTabRow
+import androidx.compose.material3.Tab
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.compose.LocalLifecycleOwner
+import androidx.lifecycle.flowWithLifecycle
+import com.github.ajalt.colormath.extensions.android.composecolor.toComposeColor
+import org.cru.godtools.shared.renderer.content.extensions.visibility
+import org.cru.godtools.shared.renderer.state.State
+import org.cru.godtools.shared.tool.parser.model.Tabs
+import org.cru.godtools.shared.tool.parser.model.stylesParent
+
+private val CORNER_RADIUS = 10.dp
+private val DEFAULT_SELECTED_COLOR = Color.DarkGray
+private val DEFAULT_UNSELECTED_COLOR = Color.White
+
+@Composable
+internal fun RenderTabs(tabs: Tabs, state: State, modifier: Modifier = Modifier) {
+    val invisible by remember(tabs, state) {
+        tabs.isInvisibleFlow(state)
+    }.collectAsState(tabs.isInvisible(state))
+
+    val lifecycleOwner = LocalLifecycleOwner.current
+    val borderColor: Color = tabs.stylesParent?.primaryColor?.toComposeColor() ?: DEFAULT_SELECTED_COLOR
+
+    var selectedIndex = remember { mutableIntStateOf(0) }
```

Comment:
this will be easier to work with if you do this instead:
```suggestion
    var selectedIndex by remember { mutableIntStateOf(0) }
```